Sources/Quick/QuickSpec.swift

```diff
@@ -68,25 +68,30 @@ open class QuickSpec: QuickSpecBase {
     }
 
     private static func addInstanceMethod(for example: Example, classSelectorNames selectorNames : inout Set<String>) -> Selector {
-        let block: @convention(block) (QuickSpec) -> Void = { spec in
+        let block: @convention(block) (QuickSpec, @escaping () -> Void) -> Void = { spec, completionHandler in
             spec.example = example
-            example.run()
+            Task {
+                await example.run()
+                completionHandler()
+            }
         }
         let implementation = imp_implementationWithBlock(block as Any)
 
         let originalName = example.name.c99ExtendedIdentifier
         var selectorName = originalName
         var index: UInt = 2
 
-        while selectorNames.contains(selectorName) {
+        while selectorNames.contains(selectorName.appending(":")) {
```

Comment:
Now that all the test names take in a single argument, we need to also append the ":" when we check for existing selectors.